### PR TITLE
Loop transformation refactoring

### PR DESCRIPTION
**Pull request by BritikovKI**
Some parts of loop transformation are moved to the Utils.
It is done to generalise the merging of the loops/vertices in the graph

### CODE REVIEW

File: src/transformers/CommonUtils.h [hunk 1]
```diff
@@ -9,6 +9,7 @@
 
 #include "graph/ChcGraph.h"
 #include "Witnesses.h"
+#include "TransformationUtils.h"
```
Comment (blishko) [inline src/transformers/CommonUtils.h:12]: Why do you need this include?

Comment (BritikovKI) [reply src/transformers/CommonUtils.h:12]: This one is needed for `getVariablesFromEdge` function which is used in EdgeTranslate

**Comment by BritikovKI** [comment]
Btw, should we mb add CommonUtils to clang-formatter?
Wdyt?

File: src/transformers/CommonUtils.h (hunk 1 above)
Comment (blishko) [reply src/transformers/CommonUtils.h:12]: Then it should be included in `.cc` file, not in the header.

**Comment by blishko** [comment]
> Btw, should we mb add CommonUtils to clang-formatter? Wdyt?

Let's not do that now, I am thinking of merging `CommonUtils` and `TransformationUtils`.

**Comment by BritikovKI** [comment]
Ok, updated, I think if we want to merge CommonUtils and TransformationUtils it should be done separately...

**Comment by blishko** [comment]
> Ok, updated, I think if we want to merge CommonUtils and TransformationUtils it should be done separately...

Indeed.

**Review by blishko — APPROVED** [review]
LGTM!

**Pull request merged**